source/slang/slang-emit-c-like.cpp

```diff
@@ -1022,50 +376,14 @@ void CLikeSourceEmitter::emitStringLiteral(
     m_writer->emit("\"");
 }
 
-void CLikeSourceEmitter::requireGLSLExtension(String const& name)
-{
-    m_glslExtensionTracker.requireExtension(name);
-}
-
-void CLikeSourceEmitter::requireGLSLVersion(ProfileVersion version)
-{
-    if (getSourceStyle() != SourceStyle::GLSL)
-        return;
-
-    m_glslExtensionTracker.requireVersion(version);
-}
-
-void CLikeSourceEmitter::requireGLSLVersion(int version)
-{
-    switch (version)
-    {
-#define CASE(NUMBER) \
-    case NUMBER: requireGLSLVersion(ProfileVersion::GLSL_##NUMBER); break
-
-    CASE(110);
-    CASE(120);
-    CASE(130);
-    CASE(140);
-    CASE(150);
-    CASE(330);
-    CASE(400);
-    CASE(410);
-    CASE(420);
-    CASE(430);
-    CASE(440);
-    CASE(450);
-
-#undef CASE
-    }
-}
-
 void CLikeSourceEmitter::setSampleRateFlag()
 {
     m_entryPointLayout->flags |= EntryPointLayout::Flag::usesAnySampleRateInput;
 }
 
 void CLikeSourceEmitter::doSampleRateInputCheck(Name* name)
 {
+    // TODO(JS): This doesn't appear to be called from anywhere!!
```

Comment:
It almost seems like it is leftover legacy code from when we handled GLSL as a source language, but I'd want to double-check that. We should validate how the "is sample rate" bit is being set in our reflection data. Ideally it wouldn't depend on codegen like this anyway.